core/mesh/src/mesh3d.cpp

```diff
@@ -381,4 +386,14 @@ QString Mesh3d::getGMSH() const {
   return msh;
 }
 
+std::vector<QColor> Mesh3d::getColorTable() const { return colorTableVec; }
+
+QVector3D Mesh3d::getOffset() const {
+
+  return QVector3D(
+      -static_cast<float>(image3_->image()->vx * image3_->xdim() / 2.0f),
```

Comment:
don't you also need to subtract the overall offset here (which I guess is 0 in your tests)?
i.e. `-image3_->image()->tx` etc